File: logica-de-programacao/dia-03/12-escopo-funcoes-e-metodos/03-exercicio1-corrigido.cs
// 1. Sorveteria

using System;

class Sorveteria
{
    // Preço de cada sabor de sorvete
    const double PRECO_CHOCOLATE = 5, PRECO_MORANGO = 6, PRECO_FLOCOS = 4;

    static void Main()
    {
        // Quantidade de sorvetes pedidos pelo cliente
        int qtdChocolate = ObterQuantidadeSorvetes("chocolate");
        int qtdMorango = ObterQuantidadeSorvetes("morango");
        int qtdFlocos = ObterQuantidadeSorvetes("flocos");

        // Cálculo do total do pedido
        double totalChocolate = CalcularSubtotal(qtdChocolate, PRECO_CHOCOLATE);
        double totalMorango = CalcularSubtotal(qtdMorango, PRECO_MORANGO);
        double totalFlocos = CalcularSubtotal(qtdFlocos, PRECO_FLOCOS);

        double totalPedido = CalcularTotal(totalChocolate, totalMorango, totalFlocos);

        // Aplicação de desconto
        int qtdTotalSorvetes = qtdChocolate + qtdMorango + qtdFlocos;

        totalPedido = AplicarDesconto(totalPedido, qtdTotalSorvetes);

        // Verificação de promoção de cobertura gratuita
        bool coberturaGratuita = TemCoberturaGratuita(totalPedido);

        // Impressão do resumo
        ExibirResumo(totalPedido, coberturaGratuita);
    }

    // Método para obter as quantidades de sorvetes do usuário
    static int ObterQuantidadeSorvetes(string sabor)
    {
        Console.Write($"Quantos sorvetes de {sabor}? ");
        
        return int.Parse(Console.ReadLine());
    }

    // Método para calcular subtotais por sabor de sorvete
    static double CalcularSubtotal(int quantidade, double preco)
    {
        return quantidade * preco;
    }

    // Método para calcular o total de todos os sorvetes
    static double CalcularTotal(double subtotalChocolate, double subtotalMorango, double subtotalFlocos)
    {
        return subtotalChocolate + subtotalMorango + subtotalFlocos;
    }
    
    // Método para verificar regra e aplicar desconto
    static double AplicarDesconto(double total, int totalSorvetes)
    {
        if (totalSorvetes > 5) // Mais do que 5 sorvetes tem desconto de 10%
        {
            total -= total / 10;
        }

        return total;
    }

    // Método para verificar regra de promoção de cobertura gratuita
    static bool TemCoberturaGratuita(double total)
    {
        if (total > 20)
        {
            return true;
        }

        return false;

        // return total > 20; // Simplificação
    }

    // Método para exibir o resumo do pedido
    static void ExibirResumo(double total, bool coberturaGratuita)
    {
        if (coberturaGratuita)
        {
            Console.WriteLine($"Total do pedido: R$ {total:0.00} e com cobertura gratuita!");
        }
        else
        {
            Console.WriteLine($"Total do pedido: R$ {total:0.00}.");
        }
    }
}

File: MM5 Front/curso-adaptacao-js-edicao-2025/dia-02/09-funcoes/03-exercicio1-corrigido.js
// Preço de cada sabor de sorvete (Constantes)
const PRECO_CHOCOLATE = 5.00;
const PRECO_MORANGO = 6.00;
const PRECO_FLOCOS = 4.00;

// Simula a entrada do usuário, retornando o valor fixo do boilerplate
function obterQuantidadeSorvetes(sabor) {
    let quantidade = 0;
    
    // Valores de simulação para replicar o pedido original (3, 4, 1)
    if (sabor === "chocolate") {
        quantidade = 3;
    } else if (sabor === "morango") {
        quantidade = 4;
    } else if (sabor === "flocos") {
        quantidade = 1;
    }
    
    console.log(`Quantos sorvetes de ${sabor}? ${quantidade}`);
    return quantidade;
}

// Calcula subtotais por sabor de sorvete
function calcularSubtotal(quantidade, preco) {
    return quantidade * preco;
}

// Soma os subtotais para obter o total do pedido
function calcularTotal(subtotalChocolate, subtotalMorango, subtotalFlocos) {
    return subtotalChocolate + subtotalMorango + subtotalFlocos;
}

/**
 * Aplica o desconto de 10% se o total de sorvetes for > 5.
 */
function aplicarDesconto(total, totalSorvetes) {
    if (totalSorvetes > 5) {
        total -= total / 10;
    }

    return total;
}

// Verifica se o pedido é elegível para a promoção de cobertura gratuita (total > R$ 20,00)
function temCoberturaGratuita(total) {
    return total > 20;
}

// Exibe o resumo final do pedido (impressão de resultados)
function exibirResumo(total, coberturaGratuita) {
    if (coberturaGratuita) {
        console.log(`\nTotal do pedido: R$ ${total.toFixed(2)} e com cobertura gratuita!`);
    } else {
        console.log(`\nTotal do pedido: R$ ${total.toFixed(2)}.`);
    }
}


// --- LÓGICA PRINCIPAL (EXECUÇÃO NO ESCOPO GLOBAL) ---

// 1. Obtenção das quantidades
// Exemplo de pedido: 3 Chocolates, 4 Morangos, 1 Flocos
const qtdChocolate = 3; 
const qtdMorango = 4;
const qtdFlocos = 1;

// 2. Cálculo dos Subtotais
const totalChocolate = calcularSubtotal(qtdChocolate, PRECO_CHOCOLATE);
const totalMorango = calcularSubtotal(qtdMorango, PRECO_MORANGO);
const totalFlocos = calcularSubtotal(qtdFlocos, PRECO_FLOCOS);

// 3. Cálculo do Total Bruto
let totalPedido = calcularTotal(totalChocolate, totalMorango, totalFlocos);

// 4. Preparação para Desconto
const qtdTotalSorvetes = qtdChocolate + qtdMorango + qtdFlocos;

// 5. Aplicação de Desconto
totalPedido = aplicarDesconto(totalPedido, qtdTotalSorvetes);

// 6. Verificação de Cobertura Gratuita
const coberturaGratuita = temCoberturaGratuita(totalPedido);

// 7. Impressão do Resumo
exibirResumo(totalPedido, coberturaGratuita);
